### PR TITLE
PAYARA-2525 Don't scan all classes when not needed.

### PR DESCRIPTION
**Pull request by arjantijms**
This removes Weld warning about using @WithAnnotations for the default
and most other cases.

### CODE REVIEW

**Comment by arjantijms** [comment]
Jenkins, quick test Payara 5 please

File: appserver/web/weld-integration/src/main/java/org/glassfish/weld/WeldDeployer.java [hunk 1]
```diff
@@ -359,6 +382,72 @@ public String getName() {
         }
     }
 
+    private Iterable<Metadata<Extension>> postProcessExtensions(Iterable<Metadata<Extension>> extensions, List<BeanDeploymentArchive> archives) {
+
+            // See if the Jersey extension that scans all classes in the bean archives is present. Normally
```
Comment (chrjohn) [inline appserver/web/weld-integration/src/main/java/org/glassfish/weld/WeldDeployer.java]: Small typo: "Normally" is duplicated

Comment (arjantijms) [reply appserver/web/weld-integration/src/main/java/org/glassfish/weld/WeldDeployer.java]: thx!

**Comment by arjantijms** [comment]
Jenkins, quick test Payara 5 please

**Comment by smillidge** [comment]
Jenkins, quick test Payara 5 please

**Pull request merged**